src/desktop/Window.cpp

```diff
@@ -1148,7 +1148,7 @@ bool CWindow::opaque() {
     if (m_fAlpha->value() != 1.f || m_fActiveInactiveAlpha->value() != 1.f)
         return false;
 
-    if (m_vRealSize->goal().floor() != m_vReportedSize)
+    if (!m_bIsX11 && m_vRealSize->goal().floor() != m_vReportedSize)
```

Comment:
dafoq is this check even for come to think of it

Reply:
This check fails when we want to tear an X11 window on a scaled monitor with `xwayland:force_zero_scaling` set to `true`. Making here an exception for the X11 windows makes it work here. We can also potentially add here the check of the  `xwayland:force_zero_scaling`  flag to only avoid it here in case the zero scaling is specified, cause this size check will always fail anyway otherwise.

Reply:
> If you ask me what this check does here in general, I have no idea. 

yes that was the question. I'd remove it altogether

Reply:
Thank you for confirming! Removing it also fixes the issue, just as intended. So I'll rewrite the commit to remove this line and this should be the fix then.